core/state/trie_prefetcher.go

```diff
@@ -376,7 +376,9 @@ func (sf *subfetcher) loop() {
 				if len(task.key) == common.AddressLength {
 					sf.trie.GetAccount(common.BytesToAddress(task.key))
 				} else {
-					sf.trie.GetStorage(sf.addr, task.key)
+					if sf.trie != nil {
+						sf.trie.GetStorage(sf.addr, task.key)
+					}
```

Comment:
this is because `OpenStorageTrie` returns its `self` parameter, which is `nil` in that case. So the prefetcher will try to prefetch some tree that is empty. You can't simply open the main tree, because that will be a different tree, and you want the main tree to be loaded with all the values needed.

So we need to make the main tree available to the prefetcher when opening the tree.

Reply:
this is fixed in https://github.com/ethereum/go-ethereum/pull/30354